File: amqp-bridge-examples/src/main/js/io/vertx/example/amqp/sender.js
var AmqpBridge = require("vertx-amqp-bridge-js/amqp_bridge");
var bridge = AmqpBridge.create(vertx);
var count = 1;

// Start the bridge, then use the event loop thread to process things thereafter.
bridge.start("localhost", 5672, function (res, res_err) {
  if (res_err !== null) {
    console.log("Bridge startup failed: " + res_err);
    return;
  }

  // Set up a producer using the bridge, send a message with it.
  var producer = bridge.createProducer("myAmqpAddress");

  // Schedule sending of a message every second
  console.log("Producer created, scheduling sends.");
  vertx.setPeriodic(1000, function (v) {
    var amqpMsgPayload = {
      "body" : "myStringContent" + count
    };

    producer.send(amqpMsgPayload);

    console.log("Sent message: " + count++);
  });
});

File: amqp-bridge-examples/src/main/ruby/io/vertx/example/amqp/sender.rb
require 'vertx-amqp-bridge/amqp_bridge'

bridge = VertxAmqpBridge::AmqpBridge.create($vertx)
@count = 1

# Start the bridge, then use the event loop thread to process things thereafter.
bridge.start("localhost", 5672) { |res_err,res|
  if (res_err != nil)
    puts "Bridge startup failed: #{res_err}"
    return
  end

  # Set up a producer using the bridge, send a message with it.
  producer = bridge.create_producer("myAmqpAddress")

  # Schedule sending of a message every second
  puts "Producer created, scheduling sends."
  $vertx.set_periodic(1000) { |v|
    amqpMsgPayload = { "body" => "myStringContent#{@count}" }

    producer.send(amqpMsgPayload)

    puts "Sent message: #{@count}"
    @count+=1
  }
}
